halo2_proofs/src/circuit/floor_planner/v1/strategy.rs

```diff
@@ -211,7 +211,23 @@ pub fn slot_in_biggest_advice_first(
             .count();
         // Sort by advice area (since this has the most contention).
         advice_cols * shape.row_count()
-    });
+    };
+
+    // This used to incorrectly use `sort_unstable_by_key` with non-unique keys, which is
+    // nondeterministic (in particular, it differs between 32-bit and 64-bit platforms).
```

Comment:
Inconsistent with the CHANGELOG description
```suggestion
    // This used to incorrectly use `sort_unstable_by_key` with non-unique keys, which is
    // deterministic but unstable (in particular, it differs between 32-bit and 64-bit platforms).
```

Reply:
"Unstable" is ambiguous here. "Nondeterministic" is correct in the sense that it's dependent on something that is not an explicit input to the function (I've always thought of platform dependence as a form of nondeterminism for that reason).
```suggestion
    // This used to incorrectly use `sort_unstable_by_key` with non-unique keys, which gave
    // output that differed between 32-bit and 64-bit platforms, and potentially between Rust
    // versions.
```